src/server/generic_family.cc

```diff
@@ -1509,6 +1509,32 @@ OpStatus GenericFamily::OpMove(const OpArgs& op_args, string_view key, DbIndex t
   return OpStatus::OK;
 }
 
+void GenericFamily::RandomKey(CmdArgList args, ConnectionContext* cntx) {
+  const static size_t kNumEntries = 15;
+  const static size_t kMaxEntriesPerShard = 2;
+
+  StringVec keys;
+  keys.reserve(kNumEntries);  // garantees that vector won't scale up on another thread
```

Comment:
This is the cause of the crash (probably). You need to `resize()` instead of `reserve()`, as `reserve()` does not initialize the objects (as they are not created).

Reply:
`resize()` initializes the items on the vector, and that was not what I was trying to do here.

Reply:
You're right. `resize()` is not what you want here, although it's a common pattern we use.
The issue you're experiencing is because you're calling `push_back()` on the same `vector` from multiple threads.
We usually initialize a vector with size == # thread, and then let each thread handle its own element. For example: https://github.com/dragonflydb/dragonfly/blob/6fce3fca9fd3b29e7422c9cc4c1a8164496e44ad/src/server/zset_family.cc#L1201-L1208
I'd propose using a `vector<StringVec>` here, call `resize()` (or just the c'tor that resizes) with the thread count like we do there, and then pass each `OpScan` below its own element within the vector. You could indeed `reserve()` each element inside the vector like you do here.

Reply:
I've implemented as suggested! It seems to have worked 🎯